edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/publisher/EventHubMetricsUpload.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Publisher
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class EventHubMetricsUpload : IMetricsPublisher
+    {
+        readonly ModuleClient moduleClient;
+        static readonly ILogger Log = Logger.Factory.CreateLogger<LogAnalyticsUpload>();
+
+        public EventHubMetricsUpload(ModuleClient moduleClient)
+        {
+            this.moduleClient = Preconditions.CheckNotNull(moduleClient, nameof(moduleClient));
+        }
+
+        public async Task PublishAsync(IEnumerable<Metric> metrics, CancellationToken cancellationToken)
+        {
+            try
+            {
+                Message metricsMessage = new Message(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(metrics)));
```

Comment:
what happens if metrics is null, passing to JsonConvert.SerializeObject?

Reply:
I've added a preconditions null check, but what do you think we do if the IEnumerable contains no metrics? I feel like there should be a check for this. Do you agree?